mlflow/pyspark/ml/__init__.py

```diff
@@ -0,0 +1,282 @@
+import logging
+from pkg_resources import resource_filename
+
+import mlflow
+from mlflow.utils import chunk_dict, truncate_dict
+from mlflow.utils.annotations import experimental
+from mlflow.utils.autologging_utils import (
+    autologging_integration,
+    safe_patch,
+    try_mlflow_log,
+)
+
+
+_logger = logging.getLogger(__name__)
+
+
+def _get_fully_qualified_class_name(instance):
+    return instance.__class__.__module__ + "." + instance.__class__.__name__
+
+
+def _read_log_model_allowlist_from_file(allowlist_file):
+    allowlist = set()
+    with open(allowlist_file) as f:
+        for line in f:
+            stripped = line.strip()
+            is_blankline_or_comment = stripped == "" or stripped.startswith("#")
+            if not is_blankline_or_comment:
+                allowlist.add(stripped)
+    return allowlist
+
+
+def _read_log_model_allowlist(spark_session):
+    """
+    Reads the module allowlist and returns it as a set.
+    """
+    allowlist_file = spark_session.sparkContext._conf.get(
+        "spark.databricks.mlflow.autolog.log_model.allowlistFile", None
```

Comment:
```suggestion
        "spark.mlflow.pysparkml.autolog.logModelAllowlistFile", None
```

Reply:
Can we rename this conf here and elsewhere? The suggestion here is mainly aimed at clarifying that this is for pyspark.ml autologging and not Databricks-specific :)